File: gerar-pdf-logins.js
const { jsPDF } = require('jspdf');
const autoTable = require('jspdf-autotable');

const alunos = [
  { nome: "DAVI DE MATOS FONSECA", ra: "000111738164-X", turma: "2ª A" },
  { nome: "GUSTAVO VIEIRA DA SILVA", ra: "000112274129-7", turma: "2ª B" },
  { nome: "JOAO VICTOR PEREIRA RODRIGUES", ra: "000113380121-3", turma: "2ª B" },
  { nome: "KERRISON LIMA RIBEIRO", ra: "000113183044-1", turma: "2ª B" },
  { nome: "ALANA GAMA DE ARAUJO", ra: "000112164941-5", turma: "2ª D" },
  { nome: "GUILHERME BIZARRIAS DE SOUZA", ra: "000114183041-3", turma: "2ª D" },
  { nome: "ISABELLY DA SILVA BESSA", ra: "000111732839-9", turma: "2ª D" },
  { nome: "ISABELLY RODRIGUES GOES", ra: "000114071847-2", turma: "2ª D" },
  { nome: "DAVID LOPES RODRIGUES COSTA", ra: "000110391065-6", turma: "2ª E" },
  { nome: "EMANUELLY MARQUES BATISTA", ra: "000114079075-4", turma: "2ª E" },
  { nome: "JEFFERSON BORGES BELMIRO DOS SANTOS", ra: "000110414712-9", turma: "2ª E" },
  { nome: "JONATHAN LIMA DE JESUS", ra: "000111878938-6", turma: "2ª E" },
  { nome: "RAFAELLA VITORIA RIOS ALVES", ra: "000124489097-2", turma: "2ª E" },
  { nome: "ANA CAROLINA DA SILVA DIAS", ra: "000110385221-8", turma: "3ª B" },
  { nome: "ANNY RAIANY MARCELINO DA SILVA", ra: "000112550639-8", turma: "3ª B" },
  { nome: "GUSTAVO HENRIQUE DOS SANTOS DE JESUS", ra: "000111878084-X", turma: "3ª B" },
  { nome: "RENAN ARTHUR INNOCENCIO DE MORAES", ra: "000110403329-X", turma: "3ª B" },
  { nome: "GUILHERME PAIN DE SOUZA", ra: "000112183892-3", turma: "3ª B" },
  { nome: "GEOVANNA DOS SANTOS SILVA", ra: "000111844942-3", turma: "3ª B" },
  { nome: "MARIA LUISA DA SILVA XAVIER", ra: "000113892805-7", turma: "3ª B" },
  { nome: "MARCELO WENDER MIRANDA RODRIGUES COSTA", ra: "000113097182-X", turma: "3ª C" },
  { nome: "KETHELLYN DOS SANTOS CAJANO", ra: "000112360623-7", turma: "3ª C" },
  { nome: "KAUE DAS NEVES SOUSA DA SILVA", ra: "000110358351-7", turma: "3ª C" },
  { nome: "JOSE HENRIQUE FERREIRA PEREIRA DA SILVA", ra: "000110413405-6", turma: "3ª C" },
  { nome: "RENAN SILVA SOUZA", ra: "000111838874-4", turma: "3ª D" },
  { nome: "JOANNA AZEVEDO DOS SANTOS", ra: "000110379588-0", turma: "3ª D" },
  { nome: "RAYSSA RAKELLY RODRIGUES PORTAS", ra: "000111836580-X", turma: "3ª D" },
  { nome: "GLORIA STEPHANY PEREIRA DA SILVA", ra: "000111844999-X", turma: "3ª D" },
  { nome: "JULIA DE LIMA PINHEIRO", ra: "000113269559-4", turma: "3ª D" },
  { nome: "LUCAS VINICIUS DA SILVA SOUZA", ra: "000112793347-4", turma: "3ª E" },
  { nome: "RUAN ALDO FERREIRA DO AMARAL", ra: "000111878754-7", turma: "3ª E" },
  { nome: "DOUGLAS FREIRE DE MELO", ra: "000113493367-8", turma: "3ª E" },
  { nome: "MARCELLA DELAMARE VIEIRA GARCIA", ra: "000112884576-3", turma: "3ª E" },
  { nome: "BEATRYZ GONCALVES DA SILVA BRUM", ra: "000112606661-8", turma: "3ª E" },
  { nome: "GABRIELE DA SILVA SANTOS", ra: "000111836491-0", turma: "3ª E" },
  { nome: "MARIA LUIZA DE JESUS PEREIRA", ra: "0001093658058", turma: "3ª E" },
  { nome: "HELOISA HELENA SOUSA DE LIMA", ra: "0001124907403", turma: "3ª E" },
  { nome: "JULIA ALVES LINHARES", ra: "0001132689028", turma: "3ª B" },
  { nome: "GUSTAVO DANIEL PACHECO", ra: "0001103290241", turma: "3ª B" }
];

const admins = [
  { nome: "Carlos", email: "carlos@adm.com", senha: "carlos123" },
  { nome: "Ivone", email: "ivone@adm.com", senha: "ivone123" }
];

const doc = new jsPDF();

// Cabeçalho
 doc.setFontSize(20);
doc.setFont('helvetica', 'bold');
doc.text('E.E. NANCY DE OLIVEIRA FIDALGO', 105, 20, { align: 'center' });

doc.setFontSize(12);
doc.setFont('helvetica', 'normal');
doc.text('Secretaria de Educação do Estado de São Paulo', 105, 28, { align: 'center' });
doc.line(20, 32, 190, 32);

// Título
doc.setFontSize(16);
doc.setFont('helvetica', 'bold');
doc.text('LISTA COMPLETA - ALUNOS E ADMINISTRADORES', 105, 45, { align: 'center' });

// Aviso importante
doc.setFontSize(10);
doc.setTextColor(255, 0, 0);
doc.setFont('helvetica', 'bold');
doc.text('⚠️  LOGIN E SENHA SÃO IGUAIS AO RA (com os ZEROS no início)', 105, 55, { align: 'center' });
doc.setTextColor(0, 0, 0);

// Seção de Administradores
doc.setFontSize(13);
doc.setFont('helvetica', 'bold');
doc.setTextColor(0, 51, 102);
doc.text('ADMINISTRADORES DO SISTEMA', 20, 70);
doc.setTextColor(0, 0, 0);

const adminData = admins.map(a => [a.nome, a.email, a.senha, 'ADM']);

autoTable(doc, {
  startY: 75,
  head: [['Nome', 'Email (Login)', 'Senha', 'Perfil']],
  body: adminData,
  theme: 'grid',
  headStyles: { fillColor: [0, 51, 102] },
  styles: { fontSize: 9, cellPadding: 3 },
  columnStyles: {
    0: { cellWidth: 40 },
    1: { cellWidth: 60 },
    2: { cellWidth: 40 },
    3: { cellWidth: 30, halign: 'center' }
  }
});

// Seção de Alunos
let finalY = doc.lastAutoTable.finalY + 10;
doc.setFontSize(13);
doc.setFont('helvetica', 'bold');
doc.setTextColor(0, 51, 102);
doc.text(`ALUNOS (${alunos.length} total)`, 20, finalY);
doc.setTextColor(0, 0, 0);

const alunosOrdenados = [...alunos].sort((a, b) => {
  if (a.turma !== b.turma) return a.turma.localeCompare(b.turma);
  return a.nome.localeCompare(b.nome);
});

const alunoData = alunosOrdenados.map(a => [
  a.nome,
  a.turma,
  a.ra,
  a.ra
]);

autoTable(doc, {
  startY: finalY + 5,
  head: [['Nome do Aluno', 'Turma', 'Login (RA)', 'Senha (RA)']],
  body: alunoData,
  theme: 'grid',
  headStyles: { fillColor: [0, 102, 51] },
  styles: { fontSize: 8, cellPadding: 2 },
  columnStyles: {
    0: { cellWidth: 80 },
    1: { cellWidth: 20, halign: 'center' },
    2: { cellWidth: 40, halign: 'center', fontStyle: 'bold' },
    3: { cellWidth: 40, halign: 'center', fontStyle: 'bold' }
  },
  alternateRowStyles: { fillColor: [245, 245, 245] }
});

// Rodapé
finalY = doc.lastAutoTable.finalY + 10;
doc.setFontSize(9);
doc.setTextColor(255, 0, 0);
doc.setFont('helvetica', 'bold');
doc.text('IMPORTANTE: Sempre digite os ZEROS no início do RA ao fazer login!', 105, finalY, { align: 'center' });

doc.setTextColor(0, 0, 0);
doc.setFont('helvetica', 'normal');
doc.setFontSize(8);
doc.text(`Documento gerado em: ${new Date().toLocaleDateString('pt-BR')} - PortãoEdu Sistema de Controle`, 105, finalY + 10, { align: 'center' });
doc.text('Este documento é de uso interno da secretaria escolar.', 105, finalY + 15, { align: 'center' });

// Rendey Class no rodapé
doc.setFontSize(7);
doc.setTextColor(100, 100, 100);
doc.text('Sistema desenvolvido por Rendey Class - www.rendey.app', 105, 290, { align: 'center' });

doc.save('LISTA_COMPLETA_ALUNOS_E_ADM.pdf');
console.log('✅ PDF gerado com sucesso: LISTA_COMPLETA_ALUNOS_E_ADM.pdf');

File: gerar_pdf_logins.py
#!/usr/bin/env python3
"""
Gerador de PDF - Lista de Logins PortãoEdu
E.E. Nancy de Oliveira Fidalgo
"""

from fpdf import FPDF
from datetime import datetime

class PDF(FPDF):
    def header(self):
        # Logo/Header
        self.set_font('Arial', 'B', 16)
        self.set_text_color(59, 130, 246)  # Azul primary
        self.cell(0, 10, 'PORTÃOEDU - SISTEMA DE CONTROLE DE ENTRADA', 0, 1, 'C')
        self.set_font('Arial', '', 12)
        self.set_text_color(100, 100, 100)
        self.cell(0, 8, 'E.E. Nancy de Oliveira Fidalgo', 0, 1, 'C')
        self.ln(5)
        
        # Linha separadora
        self.set_draw_color(59, 130, 246)
        self.line(10, self.get_y(), 200, self.get_y())
        self.ln(5)
    
    def footer(self):
        self.set_y(-15)
        self.set_font('Arial', 'I', 8)
        self.set_text_color(128, 128, 128)
        self.cell(0, 10, f'Página {self.page_no()} | Gerado em: {datetime.now().strftime("%d/%m/%Y %H:%M")}', 0, 0, 'C')

def gerar_pdf():
    pdf = PDF()
    pdf.add_page()
    pdf.set_auto_page_break(auto=True, margin=15)
    
    # Título
    pdf.set_font('Arial', 'B', 18)
    pdf.set_text_color(31, 41, 55)
    pdf.cell(0, 12, 'LISTA DE ACESSO - LOGINS E SENHAS', 0, 1, 'C')
    pdf.ln(8)
    
    # Seção ADMINISTRAÇÃO
    pdf.set_font('Arial', 'B', 14)
    pdf.set_fill_color(59, 130, 246)
    pdf.set_text_color(255, 255, 255)
    pdf.cell(0, 10, '  ADMINISTRAÇÃO (ACESSO CARLOS / IVONE)', 0, 1, 'L', fill=True)
    pdf.ln(5)
    
    pdf.set_font('Arial', '', 11)
    pdf.set_text_color(0, 0, 0)
    
    # Tabela Admin
    pdf.set_fill_color(243, 244, 246)
    pdf.cell(60, 10, 'Nome', 1, 0, 'C', fill=True)
    pdf.cell(80, 10, 'Login', 1, 0, 'C', fill=True)
    pdf.cell(50, 10, 'Senha', 1, 1, 'C', fill=True)
    
    pdf.cell(60, 10, 'Carlos', 1, 0, 'L')
    pdf.cell(80, 10, 'carlos@adm.com', 1, 0, 'L')
    pdf.cell(50, 10, 'carlos123', 1, 1, 'L')
    
    pdf.cell(60, 10, 'Ivone', 1, 0, 'L')
    pdf.cell(80, 10, 'ivone@adm.com', 1, 0, 'L')
    pdf.cell(50, 10, 'ivone123', 1, 1, 'L')
    
    pdf.ln(10)
    
    pdf.set_font('Arial', 'B', 14)
    pdf.set_fill_color(16, 185, 129)  # Verde
    pdf.set_text_color(255, 255, 255)
    pdf.cell(0, 10, '  ALUNOS - 2º ANO (LOGIN/SENHA É O RG)', 0, 1, 'L', fill=True)
    pdf.ln(5)
    
    # Cabeçalho da tabela
    pdf.set_font('Arial', 'B', 10)
    pdf.set_fill_color(243, 244, 246)
    pdf.set_text_color(0, 0, 0)
    pdf.cell(90, 10, 'Nome do Aluno', 1, 0, 'C', fill=True)
    pdf.cell(60, 10, 'Login/Senha (RG)', 1, 0, 'C', fill=True)
    pdf.cell(40, 10, 'Turma', 1, 1, 'C', fill=True)
    
    # Dados dos alunos - 2º ANO
    pdf.set_font('Arial', '', 10)
    
    alunos_2ano = [
        ('DAVI DE MATOS FONSECA', '000111738164-X', '2ª A'),
        ('GUSTAVO VIEIRA DA SILVA', '000112274129-7', '2ª B'),
        ('JOAO VICTOR PEREIRA RODRIGUES', '000113380121-3', '2ª B'),
        ('KERRISON LIMA RIBEIRO', '000113183044-1', '2ª B'),
        ('ALANA GAMA DE ARAUJO', '000112164941-5', '2ª D'),
        ('GUILHERME BIZARRIAS DE SOUZA', '000114183041-3', '2ª D'),
        ('ISABELLY DA SILVA BESSA', '000111732839-9', '2ª D'),
        ('ISABELLY RODRIGUES GOES', '000114071847-2', '2ª D'),
        ('DAVID LOPES RODRIGUES COSTA', '000110391065-6', '2ª E'),
        ('EMANUELLY MARQUES BATISTA', '000114079075-4', '2ª E'),
        ('JEFFERSON BORGES BELMIRO', '000110414712-9', '2ª E'),
        ('JONATHAN LIMA DE JESUS', '000111878938-6', '2ª E'),
        ('RAFAELLA VITORIA RIOS ALVES', '000124489097-2', '2ª E'),
    ]
    
    fill = False
    for nome, rg, turma in alunos_2ano:
        if fill:
            pdf.set_fill_color(249, 250, 251)
        else:
            pdf.set_fill_color(255, 255, 255)
        
        pdf.cell(90, 8, nome, 1, 0, 'L', fill=fill)
        pdf.cell(60, 8, rg, 1, 0, 'C', fill=fill)
        pdf.cell(40, 8, turma, 1, 1, 'C', fill=fill)
        fill = not fill
    
    pdf.ln(5)
    
    # Seção 3º ANO - Nova página se necessário
    if pdf.get_y() > 200:
        pdf.add_page()
    
    pdf.set_font('Arial', 'B', 14)
    pdf.set_fill_color(59, 130, 246)  # Azul
    pdf.set_text_color(255, 255, 255)
    pdf.cell(0, 10, '  ALUNOS - 3º ANO (LOGIN/SENHA É O RG)', 0, 1, 'L', fill=True)
    pdf.ln(5)
    
    # Cabeçalho
    pdf.set_font('Arial', 'B', 10)
    pdf.set_fill_color(243, 244, 246)
    pdf.set_text_color(0, 0, 0)
    pdf.cell(90, 10, 'Nome do Aluno', 1, 0, 'C', fill=True)
    pdf.cell(60, 10, 'Login/Senha (RG)', 1, 0, 'C', fill=True)
    pdf.cell(40, 10, 'Turma', 1, 1, 'C', fill=True)
    
    # Dados 3º ANO
    pdf.set_font('Arial', '', 10)
    
    alunos_3ano = [
        ('ANA CAROLINA DA SILVA DIAS', '000110385221-8', '3ª B'),
        ('ANNY RAIANY MARCELINO DA SILVA', '000112550639-8', '3ª B'),
        ('GUSTAVO HENRIQUE DOS SANTOS', '000111878084-X', '3ª B'),
        ('RENAN ARTHUR INNOCENCIO', '000110403329-X', '3ª B'),
        ('GUILHERME PAIN DE SOUZA', '000112183892-3', '3ª B'),
        ('GEOVANNA DOS SANTOS SILVA', '000111844942-3', '3ª B'),
        ('MARIA LUISA DA SILVA XAVIER', '000113892805-7', '3ª B'),
        ('MARCELO WENDER MIRANDA', '000113097182-X', '3ª C'),
        ('KETHELLYN DOS SANTOS CAJANO', '000112360623-7', '3ª C'),
        ('KAUE DAS NEVES SOUSA DA SILVA', '000110358351-7', '3ª C'),
        ('JOSE HENRIQUE FERREIRA', '000110413405-6', '3ª C'),
        ('RENAN SILVA SOUZA', '000111838874-4', '3ª D'),
        ('JOANNA AZEVEDO DOS SANTOS', '000110379588-0', '3ª D'),
        ('RAYSSA RAKELLY RODRIGUES', '000111836580-X', '3ª D'),
        ('GLORIA STEPHANY PEREIRA', '000111844999-X', '3ª D'),
        ('JULIA DE LIMA PINHEIRO', '000113269559-4', '3ª D'),
        ('LUCAS VINICIUS DA SILVA', '000112793347-4', '3ª E'),
        ('RUAN ALDO FERREIRA DO AMARAL', '000111878754-7', '3ª E'),
        ('DOUGLAS FREIRE DE MELO', '000113493367-8', '3ª E'),
        ('MARCELLA DELAMARE VIEIRA', '000112884576-3', '3ª E'),
        ('BEATRYZ GONCALVES DA SILVA', '000112606661-8', '3ª E'),
        ('GABRIELE DA SILVA SANTOS', '000111836491-0', '3ª E'),
    ]
    
    for nome, rg, turma in alunos_3ano:
        if fill:
            pdf.set_fill_color(249, 250, 251)
        else:
            pdf.set_fill_color(255, 255, 255)
        
        pdf.cell(90, 8, nome, 1, 0, 'L', fill=fill)
        pdf.cell(60, 8, rg, 1, 0, 'C', fill=fill)
        pdf.cell(40, 8, turma, 1, 1, 'C', fill=fill)
        fill = not fill
    
    pdf.ln(10)
    
    # Regras de Acesso
    pdf.set_font('Arial', 'B', 14)
    pdf.set_fill_color(239, 68, 68)  # Vermelho
    pdf.set_text_color(255, 255, 255)
    pdf.cell(0, 10, '  REGRAS DE ACESSO (Turno Noturno 19h-22:45h)', 0, 1, 'L', fill=True)
    pdf.ln(5)
    
    pdf.set_font('Arial', '', 11)
    pdf.set_text_color(0, 0, 0)
    
    regras = [
        ('1ª Aula (19:00 - 19:45):', 'Entrada Liberada para todos.'),
        ('2ª Aula (19:45 - 20:30):', 'Entrada SOMENTE para alunos autorizados (lista acima).'),
        ('3ª a 5ª Aula:', 'Entrada BLOQUEADA. Aluno deve ir para a DIREÇÃO/SECRETARIA.'),
    ]
    
    for aula, regra in regras:
        pdf.set_font('Arial', 'B', 11)
        pdf.cell(60, 8, aula, 0, 0, 'L')
        pdf.set_font('Arial', '', 11)
        pdf.cell(0, 8, regra, 0, 1, 'L')
        pdf.ln(2)
    
    pdf.ln(5)
    
    # Nota no rodapé da página
    pdf.set_font('Arial', 'I', 10)
    pdf.set_text_color(100, 100, 100)
    pdf.multi_cell(0, 6, 'Nota: Geração de PDF de assinatura automática no login (para 1ª e 2ª aula).', 0, 'C')
    
    # Salvar
    output_path = r'C:\Users\RenanBezerraDosSanto\Desktop\sistema-controle-nancy\LISTA_LOGINS_PORTAOEDU.pdf'
    pdf.output(output_path)
    print(f'✅ PDF gerado com sucesso!')
    print(f'📄 Local: {output_path}')
    return output_path

if __name__ == '__main__':
    gerar_pdf()
